src/__test__/logic/calculate.test.js

```diff
@@ -0,0 +1,62 @@
+import Calculate from '../../logic/calculate';
+
+const firstObject = { total: null, next: null, operation: null };
+const secondObject = { total: '6', next: null, operation: null };
+
+describe('Calculate methods', () => {
+  it('Should be a function', () => {
+    expect(typeof Calculate).toEqual('function');
+  });
+  it('Should not be an object', () => {
+    expect(typeof Calculate).not.toEqual('object');
+  });
+  it('Should return an object', () => {
+    const result = Calculate(firstObject, 9);
+    expect(typeof result).toEqual('object');
+  });
+  it('Should total be a number', () => {
+    const result = Calculate(secondObject, 9);
+    const value = (result.total * 1);
+    expect(typeof value).toEqual('number');
+  });
+  it('Should return null object', () => {
+    const result = Calculate(firstObject, 'AC');
+    expect(result).toStrictEqual({ next: null, operation: null, total: null });
+  });
+  it('Should return a decimal value', () => {
+    const result = Calculate(firstObject, '.');
+    expect(result).toStrictEqual({ next: null, operation: null, total: '0.' });
```

Comment:
This mutation is not happening in the main app, the calculator does not allow for users to enter decimals, kindly revisit the calculator and make the tested feature reflect.